elements/lisk-validator/src/keywords/data_type.ts

```diff
@@ -60,19 +69,27 @@ const compile = (
 	}
 
 	return (
-		_data: string,
+		_data: Buffer | bigint | string | number,
 		_dataPath?: string,
 		_parentData?: object,
 		_parentDataProperty?: string | number,
 		_rootData?: object,
-	): boolean =>
+	): boolean => {
+		if (value === 'boolean') return isBoolean(_data);
+		if (value === 'bytes') return isBytes(_data as Buffer);
+		if (value === 'string') return isString(value);
```

Comment:
```suggestion
		if (value === 'string') return isString(_data);
```

Reply:
@ishantiw This was logically wrong. Pleas confirm you have unit test for this type, otherwise this should not pass.

Reply:
specifically this can pass because both `value` and `_data` are string :P so that's why it passed unit tests